ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/properties/LocalPropertyParser.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.ksql.rest.client.properties;
+
+import io.confluent.ksql.ddl.DdlConfig;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlConstants;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Objects;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigDef.ConfigKey;
+import org.apache.kafka.streams.StreamsConfig;
+
+class LocalPropertyParser implements PropertyParser {
+
+  private static final ConfigDef CONSUMER_CONFIG_DEF = getConfigDef(ConsumerConfig.class);
+  private static final ConfigDef PRODUCER_CONFIG_DEF = getConfigDef(ProducerConfig.class);
+  private static final Method PARSE_METHOD = getParseMethod();
+
+  private final PropertiesValidator validator;
+
+  LocalPropertyParser() {
+    this(new LocalPropertiesValidator());
+  }
+
+  LocalPropertyParser(final PropertiesValidator validator) {
+    this.validator = Objects.requireNonNull(validator, "validator");
+  }
+
+  @Override
+  public Object parse(final String property, final Object value) {
```

Comment:
Could we clean this up a bit more by not having the variables `def, parsedProperty` set in each `if` block, but rather doing something like:
`return validate(def, parsedProperty)` and extracting the last five lines into that method?

Reply:
Give a man an inch...